### PR TITLE
Display Duration on VSM for stages

### PR DESCRIPTION
**Pull request by rohitggarg**
As part of software maturity assessment, we would like to extract stats from GoCD about a particular repository's VSM with regards to how much time it takes to either successfully deploy to production or fail.
We also need to extract the data regarding mean time to recover. Due to these requirements, we need a feature in GoCD which can quickly display the times on hover for each stage on the VSM itself.
Thanks,
RohitG

### CODE REVIEW

File: server/webapp/WEB-INF/rails.new/app/assets/javascripts/vsm_renderer.js [hunk 1]
```diff
@@ -335,7 +335,7 @@ Graph_Renderer = function (container) {
                     gui += '" style="width:' + ((stagesWidth - (stagesCount * 4)) / stagesCount) + 'px" title="' + instance.stages[i].name + '"><span>' + instance.stages[i].name + '</span></li>'
                 }
                 else {
-                    gui += '" style="width:' + ((stagesWidth - (stagesCount * 4)) / stagesCount) + 'px" title="' + instance.stages[i].name + '"><span><span></span></span><a href="' + instance.stages[i].locator + '"><span>' + instance.stages[i].name + '</span></a></li>'
+                    gui += '" style="width:' + ((stagesWidth - (stagesCount * 4)) / stagesCount) + 'px" title="' + instance.stages[i].name + ': took ' + instance.stages[i].duration + ' seconds"><span><span></span></span><a href="' + instance.stages[i].locator + '"><span>' + instance.stages[i].name + '</span></a></li>'
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/assets/javascripts/vsm_renderer.js]: Do you think you could make this render in a human readable duration? Assume jobs can take anywhere from a few days to a few seconds to complete :)

Comment (rohitggarg) [inline server/webapp/WEB-INF/rails.new/app/assets/javascripts/vsm_renderer.js]: Nice idea! I was trying to find a place where I could keep the duration displayed continuously (without the need to hover). I will check the possibility of both and then resubmit the pr.

Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/assets/javascripts/vsm_renderer.js]: You could have that in the title attribute. Would need to hover on it though. We have been doing the title attribute with other things. Hover on the timestamps on the dashboard. 17.4 onwards.

Comment (rohitggarg) [inline server/webapp/WEB-INF/rails.new/app/assets/javascripts/vsm_renderer.js]: I have added the changes, tried to use the existing cruise time formatter script.

**Comment by varshavaradarajan** [comment]
@rohitggarg , There seem to be some rspec failures for your PR, can you take a look at them? 
https://build.gocd.io/go/pipelines/build-windows-PR/1356/build-server/1 and https://build.gocd.io/go/tab/build/detail/build-linux-PR/1691/build-server/1/rspec-split-runInstance-1

**Comment by ketan** [comment]
Closing in favor of #3472.

**Pull request closed without merging**